src/Sylius/Behat/Service/SessionManager.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Behat\Service;
+
+use Behat\Mink\Mink;
+use Symfony\Component\Security\Core\Exception\TokenNotFoundException;
+
+final class SessionManager implements SessionManagerInterface
+{
+    private Mink $mink;
+
+    private SharedStorageInterface $sharedStorage;
+
+    private SecurityServiceInterface $securityService;
+
+    private const SESSION_CHROME_HEADLESS_SECOND = 'chrome_headless_second_session';
+
+    public function __construct(Mink $mink, SharedStorageInterface $sharedStorage, SecurityServiceInterface $securityService)
+    {
+        $this->mink = $mink;
+        $this->sharedStorage = $sharedStorage;
+        $this->securityService = $securityService;
+    }
+
+    public function changeSession(): void
+    {
+        $sessionName = self::SESSION_CHROME_HEADLESS_SECOND;
+
+        $this->saveAndRestartSession($sessionName);
+
+        if ($this->sharedStorage->has(sprintf('behat_previous_session_token_%s', $sessionName))) {
```

Comment:
`sprintf('behat_previous_session_token_%s', $sessionName)` could be extracted to variable... or private function to avoid duplications and possible mistakes